src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobLister.java

```diff
@@ -26,33 +27,38 @@ public BlobLister(BlobServiceClient storageClient, ServiceConfiguration serviceC
         this.serviceConfiguration = serviceConfiguration;
     }
 
-    public List<BlobInfo> listBlobs(Predicate<BlobItem> filter) {
+    public List<BlobInfo> listBlobs(int staleTime) {
         return serviceConfiguration
             .getSourceContainers()
             .stream()
-            .map(c -> listBlobsByContainer(c, filter))
+            .map(c -> listBlobsByContainer(c, staleTime))
             .flatMap(Collection::stream)
             .collect(Collectors.toList());
     }
 
-    private List<BlobInfo> listBlobsByContainer(String containerName, Predicate<BlobItem> filter) {
+    private List<BlobInfo> listBlobsByContainer(String containerName, int staleTime) {
         return storageClient.getBlobContainerClient(containerName)
             .listBlobs()
             .stream()
-            .filter(filter)
-            .map(blob -> new BlobInfo(containerName, blob.getName(),
-                toLocalTimeZone(blob.getProperties().getCreationTime().toInstant())))
+            .filter(b -> timeFilter(b, staleTime))
+            .map(blob -> new BlobInfo(
+                    containerName,
+                    blob.getName(),
+                    toLocalTimeZone(blob.getProperties().getCreationTime().toInstant())
+                )
+            )
             .collect(Collectors.toList());
```

Comment:
oh, didn't notice before - could just flatmap straight away - just return `Stream` in that first private method and should be all good :+1: